### DISCUSSION TITLE
send me a camera screenshot at different intervals

### DISCUSSION THREAD

**Discussion by starschild86** [greghesp/ha-bambulab#458] [Ideas]
hi,
i would like to be able to have HA send me telegram notifications at different intervals including a photo from the camera.
i already have setup telegram notifications in general and would like to send one to my account with the following:
- printing is at x%
- camera snapshot attached
this could be done at various intervals, e.g. at 15%, 50%, 99% (to have a photo of print almost complete before the bed gets lower)
do you have an idea on how to do this?
best,
Nicola

**Comment by jjvelar** [comment]
I have created this automation in Home Assistant. I hope it helps:

`alias: Notificación - Bambu Lab A1 Print Progress
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.a1_xxxxxxxxxxxxxxx_print_status
    to: prepare
    id: Preparing
  - platform: state
    entity_id:
      - sensor.a1_xxxxxxxxxxxxxxx_print_status
    to: running
    id: Start
  - platform: state
    entity_id:
      - sensor.a1_xxxxxxxxxxxxxxx_print_status
    to: finish
    id: Finished
  - platform: state
    entity_id:
      - sensor.a1_xxxxxxxxxxxxxxx_print_status
    to: Failed
    id: Failed
    from: Running
  - platform: state
    entity_id:
      - sensor.a1_xxxxxxxxxxxxxxx_print_status
    to: Pause
    id: Pause
    from: Running
  - platform: template
    value_template: >-
      {{ states('sensor.a1_xxxxxxxxxxxxxxx_print_progress') | int % 25 == 0 and
      states('sensor.a1_xxxxxxxxxxxxxxx_print_progress') | int > 0 and
      states('sensor.a1_xxxxxxxxxxxxxxx_print_progress') | int < 100 }}
condition: []
action:
  - service: button.press
    metadata: {}
    data: {}
    target:
      entity_id: button.a1_xxxxxxxxxxxxxxx_force_refresh_data
  - delay:
      hours: 0
      minutes: 0
      seconds: 5
      milliseconds: 0
  - if:
      - condition: trigger
        id:
          - Finished
          - Preparing
          - Failed
          - Start
    then:
      - service: notify.telegram_jv
        data:
          title: ""
          message: ""
          data:
            photo:
              - url: >-
                  http://HOME_ASSISTANT_LOCAL_IP:PORT{{state_attr('image.a1_xxxxxxxxxxxxxxx_camera',
                  'entity_picture') }}
                caption: |-
                  *Bambu Lab A1 Print Job*
                  - Status: {{ trigger.id }}
    else:
      - service: notify.telegram_jv
        data:
          title: ""
          message: ""
          data:
            photo:
              - url: >-
                  http://HOME_ASSISTANT_LOCAL_IP:PORT{{state_attr('image.a1_xxxxxxxxxxxxxxx_camera',
                  'entity_picture') }}
                caption: >-
                  *Bambu Lab A1 Print Job*

                  - File: {{states('sensor.a1_xxxxxxxxxxxxxxx_gcode_filename')|
                  replace('_', '\\_')}}

                  - Progress:
                  {{states('sensor.a1_xxxxxxxxxxxxxxx_print_progress')}}%

                  - Remaining time: {% set minutes =
                  states('sensor.a1_xxxxxxxxxxxxxxx_remaining_time') | int %}{%
                  set days = (minutes // (24 * 60)) %}{% set hours = (minutes %
                  (24 * 60)) // 60 %}{% set remaining_minutes = minutes % 60
                  %}{% if days > 0 %}{{days}}d {% endif %}{% if hours > 0
                  %}{{hours}}h {% endif %}{{remaining_minutes}}m
mode: single`
